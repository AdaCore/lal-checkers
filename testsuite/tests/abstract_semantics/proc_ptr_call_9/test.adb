procedure Test is
   P : access function return Integer;
   X : Integer := P.all;
begin
   null;
end Test;
