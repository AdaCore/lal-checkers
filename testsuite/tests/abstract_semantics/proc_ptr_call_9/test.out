{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "assume0",
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[non-null]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "assign1": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assume0",
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[non-null]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "assign2": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assume0",
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[non-null]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "assume0": [
      {
        "trace:": [
          "assume0",
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[non-null]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "read0": [
      {
        "trace:": [
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[all-path]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "start0": [
      {
        "trace:": [
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[all-path]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ]
  },
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0",
          "assume0",
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[non-null]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "assign1": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assume0",
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[non-null]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "assign2": [
      {
        "trace:": [
          "assign0",
          "assign1",
          "assign2",
          "assume0",
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[non-null]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "assume0": [
      {
        "trace:": [
          "assume0",
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[non-null]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "read0": [
      {
        "trace:": [
          "read0",
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[all-path]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ],
    "start0": [
      {
        "trace:": [
          "start0"
        ],
        "values": {
          "$stack": "({}, 0)",
          "P": "{[all-path]}",
          "X": "[-2147483648, 2147483647]"
        }
      }
    ]
  }
}